autogpt_platform/backend/backend/blocks/slantthreed/webhook.py

```diff
@@ -0,0 +1,125 @@
+        )
+
+    class Output(Slant3DTriggerBase.Output):
+        status: str = SchemaField(description="The new status of the order")
+        tracking_number: str = SchemaField(
+            description="The tracking number for the shipment"
+        )
+        carrier_code: str = SchemaField(description="The carrier code (e.g., 'usps')")
+
+    def __init__(self):
+        super().__init__(
+            id="8a74c2ad-0104-4640-962f-26c6b69e58cd",
+            description=(
+                "This block triggers on Slant3D order status updates and outputs "
+                "the event details, including tracking information when orders are shipped."
+            ),
+            # All webhooks are currently subscribed to for all orders. This works for self hosted, but not for cloud hosted
+            disabled=settings.Settings().config.behave_as == BehaveAs.CLOUD,
+            categories={BlockCategory.DEVELOPER_TOOLS},
+            input_schema=self.Input,
+            output_schema=self.Output,
+            webhook_config=BlockWebhookConfig(
+                provider="slant3d",
+                webhook_type="orders",  # Only one type for now
+                resource_format="",  # No resource format needed
+                event_filter_input="events",
+                event_format="order.{event}",
+            ),
+            test_input={
+                "credentials": TEST_CREDENTIALS_INPUT,
+                "events": {"shipped": True},
+                "payload": {
+                    "orderId": "1234567890",
+                    "status": "SHIPPED",
+                    "trackingNumber": "ABCDEF123456",
+                    "carrierCode": "usps",
+                },
+            },
+            test_credentials=TEST_CREDENTIALS,
+            test_output=[
+                (
+                    "payload",
+                    {
+                        "orderId": "1234567890",
+                        "status": "SHIPPED",
+                        "trackingNumber": "ABCDEF123456",
+                        "carrierCode": "usps",
+                    },
+                ),
+                ("order_id", "1234567890"),
+                ("status", "SHIPPED"),
+                ("tracking_number", "ABCDEF123456"),
+                ("carrier_code", "usps"),
+            ],
+        )
+
+    def run(self, input_data: Input, **kwargs) -> BlockOutput:  # type: ignore
+        yield from super().run(input_data, **kwargs)
+
+        try:
+            # Extract and normalize values from the payload
+            yield "status", input_data.payload["status"]
+            yield "tracking_number", input_data.payload["trackingNumber"]
+            yield "carrier_code", input_data.payload["carrierCode"]
+        except KeyError as e:
+            error_msg = f"Missing required field in webhook payload: {e}"
+            yield "error", error_msg
+            raise RuntimeError(error_msg)
```

Comment:
Redundant with check in `validate_payload`

Reply:
Update: not redundant if `payload` has default value `{}`. And I assume it has that default value so it doesn't completely break a graph if there are multiple entrypoints. We really need a structural solution for that...

Reply:
Update: #8853 makes `payload = {}` unnecessary.

Reply:
yeeted the check